packages/react/src/auto/polaris/tableCells/PolarisAutoTableTagCell.tsx

```diff
@@ -1,32 +1,61 @@
-import { InlineStack, Tag } from "@shopify/polaris";
-import React, { useMemo } from "react";
+import { InlineStack, Tag, Text } from "@shopify/polaris";
+import React, { useEffect, useMemo, useRef, useState } from "react";
 import type { RoleAssignmentsValueType } from "../../../utils.js";
 import { isRoleAssignmentsArray } from "../../../utils.js";
 
+const MAX_TAGS_LENGTH = 5;
+
 export const PolarisAutoTableTagCell = (props: { value: string | string[] | RoleAssignmentsValueType[] }) => {
   const { value } = props;
+  const [showMore, setShowMore] = useState(false);
+  const ref = useRef<HTMLDivElement>(null);
+
+  const { tags, originalLength } = useMemo(() => {
+    let formattedTags: string[] = [];
 
-  const tags = useMemo(() => {
     if (Array.isArray(value)) {
       if (isRoleAssignmentsArray(value)) {
-        return value.map((role) => role.name);
+        formattedTags = value.map((role) => role.name);
       } else if (typeof value[0] === "string") {
-        return value;
-      } else {
-        return [];
+        formattedTags = value;
       }
     } else {
-      return [value];
+      formattedTags = [value];
     }
+
+    return {
+      tags: formattedTags.slice(0, MAX_TAGS_LENGTH),
+      originalLength: formattedTags.length,
+    };
   }, [value]);
 
+  useEffect(() => {
+    if (ref.current) {
+      setShowMore(ref.current.scrollWidth > ref.current.clientWidth);
+    }
+  }, [tags]);
+
+  const showMoreComponent = <Text as="span">...</Text>;
+
   return (
     <InlineStack gap="100" wrap={false}>
-      {tags.map((tag) => (
-        <Tag key={tag} accessibilityLabel={`tag-${tag}`}>
-          {tag}
-        </Tag>
-      ))}
+      <div
+        style={{
+          width: showMore ? "calc(100% - 20px)" : "auto",
+          overflowX: "hidden",
+        }}
+        ref={ref}
+      >
+        <InlineStack gap="100" wrap={false}>
+          {tags.map((tag) => (
+            <Tag key={tag} accessibilityLabel={`tag-${tag}`}>
```

Comment:
`accessibilityLabel` is the tooltip and when it is undefined, it will be the complete untruncated tag content. Could you remove it from here?